ui/build/build.types.js

```diff
@@ -62,22 +62,36 @@ function convertTypeVal (type, def, required) {
     }
   }
 
-  if (typeMap.has(t)) {
-    return typeMap.get(t)
+  if (typeMap.has(type)) {
+    return typeMap.get(type)
   }
 
-  if (fallbackComplexTypeMap.has(t)) {
+  if (fallbackComplexTypeMap.has(type)) {
     if (def.definition) {
-      const propDefinitions = getPropDefinitions({ propDefs: def.definition, required, docs: true })
+      const propDefinitions = getPropDefinitions({ definitions: def.definition, required })
       const lines = []
-      propDefinitions.forEach(p => lines.push(...p.split('\n')))
-      return propDefinitions && propDefinitions.length > 0 ? `{\n        ${ lines.join('\n        ') } }${ t === 'Array' ? '[]' : '' }` : fallbackComplexTypeMap.get(t)
+      propDefinitions.forEach(propDef => writeLines(lines, propDef, 2))
+
+      if (lines.length > 0) {
+        return `{ ${ lines.join('') } }${ type === 'Array' ? '[]' : '' }`
+      }
     }
 
-    return fallbackComplexTypeMap.get(t)
+    return fallbackComplexTypeMap.get(type)
   }
 
-  return t
+  if (type === 'Function') {
+    // FIXME: Find a better way to handle this, figure out if there are other cases as well
+    // Set paramsRequired to false for the function returned from Notify.create(...)
```

Comment:
Let's add a "requiredParams: false" to the API on the same level as "params" in such cases.

Reply:
Otherwise looks really really good!